### PR TITLE
test(grid): refactor for cypress tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Refactor Cypress tests for the `Grid` component to use new cypress-component-react framework for testing.
Add `grid.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `grid.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/grid/grid.test.js [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+        ViewportSize("default");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        } else {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
```
Comment (sianford) [inline src/components/grid/grid.test.js:78]: What's the difference between this assertion in the `else` and the case in the `if`? They both appear to be checking the same things. Perhaps the second `gridPod(index).should...` from the `if` and the first `gridPod(index).should...` in the else are not required?

File: src/components/grid/grid.test.js [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        }
+      }
+    );
+
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
```
Comment (sianford) [inline src/components/grid/grid.test.js:93]: Should this read "extra small"? Also, should there then be a case for checking "small" after the "extra small" one?

File: src/components/grid/grid.test.js [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to medium and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("medium");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
```
Comment (sianford) [inline src/components/grid/grid.test.js:127]: Same query here about the possible duplication of assertion in the `if` and `else` as in line 78 onwards.

File: src/components/grid/grid.test.js [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to large and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("large");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
```
Comment (sianford) [inline src/components/grid/grid.test.js:155]: Same query here about the possible duplication of assertion in the `if` and `else` as in line 78 onwards.

File: src/components/grid/grid.test.js [hunk 5]
```diff
@@ -0,0 +1,360 @@
+import * as React from "react";
+import GridContainer from "./grid-container/grid-container.component";
+import GridItem from "./grid-item/grid-item.component";
+import { pod, gridPod, gridComponent } from "../../../cypress/locators/grid";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const ViewportSize = (sizeOfViewport) => {
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:7]:
```suggestion
const viewportSize = (sizeOfViewport) => {
```

**Review by DlgSHi — COMMENTED** [review]
lgtm apart from one small comment and those ones from @sianford

File: src/components/grid/grid.test.js [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+        }
+      }
+    );
+
+    it.each([
+      [1, "1 /1", "1 / 13", 1179],
+      [2, "2 / 3", "1 / 6", 477],
+      [3, "4 / 5", "7 / 13", 577],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
```
Comment (sianford) [inline src/components/grid/grid.test.js:201]: The viewport size should be changed here to `medium`.

File: src/components/grid/grid.test.js [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+        }
+      }
+    );
+
+    it.each([
+      [1, "1 /1", "1 / 13", 1824],
+      [2, "2 / 3", "1 / 6", 746],
+      [3, "4 / 5", "7 / 13", 900],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
```
Comment (sianford) [inline src/components/grid/grid.test.js:235]: The viewport size should be changed here to `large`.

File: src/components/grid/grid.test.js [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+        }
+      }
+    );
+
+    it.each([
+      [1, "1 /1", "1 / 13", 1826],
+      [2, "2 / 3", "1 / 6", 737],
+      [3, "4 / 5", "7 / 13", 893],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
```
Comment (sianford) [inline src/components/grid/grid.test.js:269]: The viewport size should be changed here to `extra large`.

**Review by sianford — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Pull request merged**